netebpfext/net_ebpf_ext_sock_addr.c

```diff
@@ -423,7 +841,478 @@ net_ebpf_extension_sock_addr_authorize_connection_classify(
+    }
+
+    connection_context_original = (net_ebpf_extension_connection_context_t*)ExAllocatePoolUninitialized(
+        NonPagedPoolNx, sizeof(net_ebpf_extension_connection_context_t), NET_EBPF_EXTENSION_POOL_TAG);
+    if (connection_context_original == NULL) {
+        status = STATUS_INSUFFICIENT_RESOURCES;
+        goto Exit;
+    }
+    memset(connection_context_original, 0, sizeof(net_ebpf_extension_connection_context_t));
+
+    connection_context_redirected = (net_ebpf_extension_connection_context_t*)ExAllocatePoolUninitialized(
+        NonPagedPoolNx, sizeof(net_ebpf_extension_connection_context_t), NET_EBPF_EXTENSION_POOL_TAG);
+    if (connection_context_redirected == NULL) {
+        status = STATUS_INSUFFICIENT_RESOURCES;
+        goto Exit;
+    }
+    memset(connection_context_redirected, 0, sizeof(net_ebpf_extension_connection_context_t));
+
+    // Fetch redirect state.
+    redirect_state = FwpsQueryConnectionRedirectState(incoming_metadata_values->redirectRecords, redirect_handle, NULL);
+    if (redirect_state == FWPS_CONNECTION_REDIRECTED_BY_SELF ||
+        redirect_state == FWPS_CONNECTION_PREVIOUSLY_REDIRECTED_BY_SELF) {
+        NET_EBPF_EXT_LOG_MESSAGE_UINT64_UINT64(
+            NET_EBPF_EXT_TRACELOG_LEVEL_ERROR,
+            NET_EBPF_EXT_TRACELOG_KEYWORD_SOCK_ADDR,
+            "Connection redireced by self, ignoring",
+            filter->filterId,
+            (uint64_t)sock_addr_ctx->compartment_id);
+
+        // We have already looked at this connection. Permit and exit. Populate connection
+        // contexts for this new connection, so that AUTH classify can permit this connection.
+        _net_ebpf_ext_connection_context_initialize_key(
+            sock_addr_ctx,
+            incoming_metadata_values->transportEndpointHandle,
+            true, /* original */
+            v4_mapped,
+            &connection_context_original->key);
+        _net_ebpf_ext_connection_context_initialize_key(
+            sock_addr_ctx,
+            incoming_metadata_values->transportEndpointHandle,
+            false, /* original */
+            v4_mapped,
+            &connection_context_redirected->key);
+
+        verdict = BPF_SOCK_ADDR_VERDICT_PROCEED;
+        redirected = true;
+
+        goto CreateContext;
+    }
+
+    // Acquire classify handle.
+    status = FwpsAcquireClassifyHandle((void*)classify_context, 0, &classify_handle);
+    if (!NT_SUCCESS(status)) {
+        NET_EBPF_EXT_LOG_NTSTATUS_API_FAILURE_UINT64_UINT64(
+            NET_EBPF_EXT_TRACELOG_KEYWORD_SOCK_ADDR,
+            "FwpsAcquireClassifyHandle",
+            status,
+            filter->filterId,
+            (uint64_t)sock_addr_ctx->compartment_id);
+
+        goto Exit;
+    }
+    classify_handle_acquired = true;
+
+    if (v4_mapped) {
+        sock_addr_ctx->family = AF_INET;
+        const uint8_t* v4_ip = IN6_GET_ADDR_V4MAPPED((IN6_ADDR*)&sock_addr_ctx->user_ip6);
+        uint32_t local_v4_ip = *((uint32_t*)v4_ip);
```

Comment:
Here the prefix `local` is confusing as it seems like local from network host prespective. But here it actually means a local variable. Please find a different name.